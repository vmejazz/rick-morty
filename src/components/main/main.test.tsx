import React from "react";
import { render } from "@testing-library/react";
import Main from "./main";
jest.mock(
  "../list-of-characters/list-of-characters.tsx",
  () => "ListOfCharacters"
);

test("Render Main component", () => {
  const tree = render(<Main />);
  expect(tree).toMatchSnapshot();
});
