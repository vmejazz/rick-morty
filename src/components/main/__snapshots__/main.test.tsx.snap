// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-AxjAm hqLGGC"
      >
        <h1
          class="sc-AxirZ inaZIU"
        >
          Test page Rick and Morty
        </h1>
        <div>
          <input
            class="sc-AxiKw bRNXnZ"
            placeholder="Find your rick"
            type="text"
          />
        </div>
        <listofcharacters
          pagequery="1"
          searchtext=""
        />
        <h2
          class="sc-AxhUy ESYqO"
        >
          Party
        </h2>
        <ul
          class="sc-AxhCb gWkrnZ"
        >
          <li
            class="sc-AxgMl dJzCRE"
          >
            <h3
              class="sc-AxmLO ewHHhy"
            >
              RICK
            </h3>
          </li>
          <li
            class="sc-AxgMl dJzCRE"
          >
            <h3
              class="sc-AxmLO ewHHhy"
            >
              MORTY
            </h3>
          </li>
        </ul>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-AxjAm hqLGGC"
    >
      <h1
        class="sc-AxirZ inaZIU"
      >
        Test page Rick and Morty
      </h1>
      <div>
        <input
          class="sc-AxiKw bRNXnZ"
          placeholder="Find your rick"
          type="text"
        />
      </div>
      <listofcharacters
        pagequery="1"
        searchtext=""
      />
      <h2
        class="sc-AxhUy ESYqO"
      >
        Party
      </h2>
      <ul
        class="sc-AxhCb gWkrnZ"
      >
        <li
          class="sc-AxgMl dJzCRE"
        >
          <h3
            class="sc-AxmLO ewHHhy"
          >
            RICK
          </h3>
        </li>
        <li
          class="sc-AxgMl dJzCRE"
        >
          <h3
            class="sc-AxmLO ewHHhy"
          >
            MORTY
          </h3>
        </li>
      </ul>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
